InternetHospitalWebUI/src/app/Components/Authorization/sign-up/sign-up.component.ts

```diff
@@ -10,18 +12,53 @@ import { RegistrationService } from '../../../Services/registration.service';
 })
 export class SignUpComponent implements OnInit {
 
-  constructor(private service: RegistrationService, private router: Router) { }
+  constructor(private service: RegistrationService, private validation: ImageValidationService, private router: Router) { }
 
+  defaultImage: string = '../../../assets/img/default-image.png';
+  imageUrl: string = this.defaultImage;
+  fileToUpload: File = null;
+  isImageValid: boolean = false;
   ngOnInit() {
   }
 
   onClear() {
     this.service.form.reset();
   }
 
-  onSubmit(form: NgForm) {
+  handleFileInput(file : FileList){
+    this.fileToUpload = file.item(0);
+
+    var reader = new FileReader();
+    reader.onload = (event: any) => {
+      if(event.target.readyState === FileReader.DONE)
+      {
+        var img = new Image();
+        img.src = event.target.result;
```

Comment:
There is method, which checks is file an image. It would be better to check, is file an image before loading.

Reply:
in sign-up.component.html in input (type=file) there is attribute accept="image/*" that allows only images